### PR TITLE
Fact 110 integrate idam

### PR DESCRIPTION
**Pull request by linusnorton**
### JIRA link ###

https://tools.hmcts.net/jira/browse/FACT-110

### Change description ###

Set up IDAM in preview

**Does this PR introduce a breaking change?** (check one with "x")

```
[y] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app/fact/FactApi.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { AxiosInstance } from 'axios';
+
+export class FactApi {
+
+  constructor(
+    private readonly axios: AxiosInstance,
+    private readonly logger: Logger
+  ) { }
+
+  public search(query: string): Promise<unknown[]> {
```
Comment (velimd) [inline src/main/app/fact/FactApi.ts:11]: does this need to be here?

File: src/main/modules/awilix/index.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+export class Container {
+
+  public enableFor(server: Application): void {
+
+    server.locals.container = createContainer({ injectionMode: InjectionMode.CLASSIC }).register({
+      logger: asValue(logger),
+      axios: asValue(Axios.create({ baseURL: config.get('services.api.url') })),
+      api: asClass(FactApi),
+      homeController: asClass(HomeController),
```
Comment (velimd) [inline src/main/modules/awilix/index.ts:25]: do any of these need to exist in admin?

Comment (linusnorton) [reply src/main/modules/awilix/index.ts:25]: yeah, they've been implemented

File: src/main/app/fact/FactApi.ts (hunk 1 above)
Comment (linusnorton) [reply src/main/app/fact/FactApi.ts:11]: I think it might become something like getAll - do we have an endpoint for that?

**Review by velimd — APPROVED** [review]

**Pull request merged**